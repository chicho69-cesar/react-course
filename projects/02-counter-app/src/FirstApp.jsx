import PropTypes from 'prop-types'
import { Fragment } from 'react'

export const FirstApp = ({ title, subtitle, name }) => {
  return (
    <Fragment>
      <h1 data-testid="test-title">{title}</h1>
      <p>{subtitle}</p>
      <p>{name}</p>
    </Fragment>
  )
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string,
  name: PropTypes.string
}
