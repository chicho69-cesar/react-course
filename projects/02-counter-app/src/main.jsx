import './index.css'

import { StrictMode } from 'react'
import { createRoot } from 'react-dom/client'
import { CounterApp } from './CounterApp'

// import { HelloWorldApp } from './HelloWorldApp.jsx'
// import { FirstApp } from './FirstApp'

const rootElement = document.getElementById('root')
const root = createRoot(rootElement)

root.render(
  <StrictMode>
    {/* <HelloWorldApp /> */}
    
    {/* <FirstApp
      title={'Hola, soy Goku'}
      subtitle={'Soy un subtÃ­tulo'}
      name={'Goku'}
    /> */}

    <CounterApp value={10} />
  </StrictMode>,
)
