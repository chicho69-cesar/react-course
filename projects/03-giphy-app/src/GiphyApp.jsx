import { useState } from "react"
import { AddCategory } from "./components/AddCategory"
import { GiftGrid } from "./components/GiftGrid"

function GiphyApp() {
  const [categories, setCategories] = useState(['Dragon Ball'])

  const onAddCategory = (newCategory) => {
    if (categories.includes(newCategory)) return
    setCategories([newCategory, ...categories])
  }

  return (
    <>
      <h1>
        Giphy App!
      </h1>

      <AddCategory onNewCategory={onAddCategory} />

      {categories.map((category) => (
        <GiftGrid
          key={category}
          category={category}
        />
      ))}
    </>
  )
}

export default GiphyApp
