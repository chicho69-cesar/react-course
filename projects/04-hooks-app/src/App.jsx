function App() {
  return (
    <h1>
      Hooks App!!!
    </h1>
  )
}

export default App
