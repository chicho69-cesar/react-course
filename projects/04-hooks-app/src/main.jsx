import './index.css'

// import { StrictMode } from 'react'
import { createRoot } from 'react-dom/client'

// import App from './App.jsx'
// import { CounterApp } from './01-use-state/CounterApp'
// import { CounterWithCustomHook } from './01-use-state/CounterWithCustomHook'
// import SimpleForm from './02-use-effect/SimpleForm'
// import { FormWithCustomHook } from './02-use-effect/FormWithCustomHook'
// import { Coords } from './02-use-effect/Coords'
import { MultipleCustomHooks } from './03-examples/MultipleCustomHooks'
// import { FocusScreen } from './04-use-ref/FocusScreen'

createRoot(document.getElementById('root')).render(
  // <StrictMode>
    // <App />
    // <CounterApp />
    // <CounterWithCustomHook />
    // <SimpleForm />
    // <FormWithCustomHook />
    // <Coords />
    <MultipleCustomHooks />
    // <FocusScreen />
  // </StrictMode>,
)
