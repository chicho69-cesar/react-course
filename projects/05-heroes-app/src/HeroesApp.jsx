import { AuthProvider } from './auth/context'
import { AppRouter } from './router/AppRouter'

function HeroesApp() {
  return (
    <AuthProvider>
      <AppRouter />
    </AuthProvider>
  )
}

export default HeroesApp
