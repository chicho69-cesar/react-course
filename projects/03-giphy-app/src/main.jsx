import './index.css'

import { StrictMode } from 'react'
import { createRoot } from 'react-dom/client'

import GiphyApp from './GiphyApp.jsx'

const rootItem = document.getElementById('root')
const root = createRoot(rootItem)

root.render(
  <StrictMode>
    <GiphyApp />
  </StrictMode>,
)
