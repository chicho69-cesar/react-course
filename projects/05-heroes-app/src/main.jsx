import './index.css'

import { StrictMode } from 'react'
import { createRoot } from 'react-dom/client'
import { BrowserRouter } from 'react-router'

import HeroesApp from './HeroesApp.jsx'

const rootItem = document.getElementById('root')
const root = createRoot(rootItem)

root.render(
  <StrictMode>
    <BrowserRouter>
      <HeroesApp />
    </BrowserRouter>
  </StrictMode>,
)
