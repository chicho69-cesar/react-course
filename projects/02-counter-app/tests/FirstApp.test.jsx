/* eslint-disable no-undef */
import { render } from '@testing-library/react'
import { FirstApp } from '../src/FirstApp'

describe('Test on <FirstApp />', () => {
  test('should be match with the snapshot', () => {
    const title = 'Hola, soy Goku'
    const subtitle = 'Soy un subtitulo'
    const name = 'Goku'

    const { container } = render(
      <FirstApp title={title} subtitle={subtitle} name={name} />
    )

    expect(container).toMatchSnapshot()
  })

  test('should show title on h1', () => {
    const title = 'Hola, soy Goku'
    const subtitle = 'Soy un subtitulo'
    const name = 'Goku'

    const { getByText, getByTestId } = render(
      <FirstApp title={title} subtitle={subtitle} name={name} />
    )

    expect(getByText(title)).toBeTruthy()
    /* Extraemos el elemento del DOM en base a su data-testid */
    expect(getByTestId('test-title').innerHTML).toContain(title)
  })

  test('should show the subtitle send by props', () => {
    const title = 'Hola, soy Goku'
    const subtitle = 'Soy un subtitulo'
    const name = 'Goku'

    const { getAllByText } = render(
      <FirstApp title={title} subtitle={subtitle} name={name} />
    )

    expect(getAllByText(subtitle).length).toBe(2)
  })
})
