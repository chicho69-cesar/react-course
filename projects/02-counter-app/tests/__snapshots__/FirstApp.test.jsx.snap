// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <FirstApp /> should be match with the snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, soy Goku
  </h1>
  <p>
    Soy un subtitulo
  </p>
  <p>
    Soy un subtitulo
  </p>
  <p>
    Goku
  </p>
</div>
`;
